db/migrate/20190617100508_add_limit_exceeded_error_to_proxies.rb

```diff
@@ -0,0 +1,18 @@
+class AddLimitExceededErrorToProxies < ActiveRecord::Migration
+  def up
```

Comment:
Why not like this?

```ruby
class AddLimitExceededErrorToProxies < ActiveRecord::Migration
  def change
    safety_assured { add_column :proxies, :error_headers_limits_exceeded, :string, null: false, default: 'text/plain; charset=us-ascii' }
    safety_assured { add_column :proxies, :error_status_limits_exceeded, :integer, null: false, default: 429 }
    safety_assured { add_column :proxies, :error_limits_exceeded, :string, null: false, default: 'Usage limit exceeded' }
  end
end
```

I have tried it and it works:
![image](https://user-images.githubusercontent.com/11318903/59609133-21de4180-9117-11e9-9a3a-771d7417fd7f.png)

Either way we need to be careful because:
1. It affects all current records, so we need to be careful in the way of doing in SaaS
2. WE need to run the migration for our 3 DBs 😄

Reply:
ok, that could wok :) thanks!

Reply:
Well this is explicitely saying that even you are doing a dangerous migration, you ack that and want to run it in production.
We should not do that but instead run the migrations with `SAFETY_ASSURED=1` environment when running them

Reply:
I said that because I was expecting letting ops doing it in saas and then let's for on-prem actually run the migrations, but something else that we can do is actually do it non-blocking ourselves, and by that I mean doing this in the migration(s):
1. A migration that adds the column but at first with all records to null
2. In that migration, make the validation for not null and adding the default value to all records but one by one in batches and with a waiting time between batches... maybe ops should validate that this is a nice idea.

Reply:
Yes do it non blocking, NULL with default.

The correct way to do things is:

1. writing migration normally, without `safety_assured`
2. run the migration on development environment
3. if it detects a dangerous migration, it will raise an error

In case of dangerous migration detected, do the appropriate measure described https://github.com/ankane/strong_migrations#strong-migrations